components/org.wso2.carbon.identity.application.authenticator.google/src/main/java/org/wso2/carbon/identity/application/authenticator/google/Utils.java

```diff
@@ -0,0 +1,130 @@
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * This is a utility classes for Google authenticator.
+ */
+public class Utils {
+
+    private static final List<String> ISSUER = Arrays.asList("https://accounts.google.com", "accounts.google.com");
+    private static final String JWS_RS256_URI = "https://www.googleapis.com/oauth2/v3/certs";
+    private static final String JWS_ES256_URI = "https://www.gstatic.com/iap/verify/public_key-jwk";
+
+    private Utils() {
+
+    }
+
+    /**
+     * This function validates the JWT token by its content using nimbus libraries.
+     *
+     * @param idToken  String. The jwt token string.
+     * @param audience String. Authenticator client ID to check validity.
+     * @return boolean. Validity of the JWT token returned via Google One Tap.
+     * @throws AuthenticationFailedException When JWT processor throws an exception.
+     */
+    public static boolean validateGoogleJWT(String idToken, String audience)
+            throws AuthenticationFailedException {
+
+        // Setting up the processor to verify the signature.
+        ConfigurableJWTProcessor<SecurityContext> jwtProcessor = new DefaultJWTProcessor<>();
+        jwtProcessor.setJWSKeySelector(generateKeySelector(idToken));
+
+        // Process the token.
+        JWTClaimsSet claimsSet;
+        try {
+            claimsSet = jwtProcessor.process(idToken, null);
+        } catch (ParseException | BadJOSEException | JOSEException e) {
+            throw new AuthenticationFailedException(GoogleErrorConstants.ErrorMessages
+                    .JWT_PROCESS_ERROR.getCode(), e.getMessage());
+        }
+
+        // Verifying the issuers and audiences.
+        if (claimsSet != null && claimsSet.toJSONObject() != null && !claimsSet.toJSONObject().isEmpty()) {
+            return ISSUER.contains(claimsSet.getIssuer()) && claimsSet.getAudience().contains(audience);
+        }
+
+        return false;
+    }
+
+    /**
+     * This function generates the JWSKeySelector according to the hashing algorithm in the jwt token.
+     *
+     * @param idToken String The JWT token.
+     * @return JWSKeySelector<SecurityContext> The generated key selector according to the hashing algorithm.
+     * @throws AuthenticationFailedException When exceptions are thrown at parsing.
+     */
+    private static JWSKeySelector<SecurityContext> generateKeySelector(String idToken)
+            throws AuthenticationFailedException {
+
+        JWSKeySelector<SecurityContext> jwsKeySelector;
+        JWT jwt;
+        try {
+            jwt = JWTParser.parse(idToken);
+        } catch (ParseException e) {
+            throw new AuthenticationFailedException(GoogleErrorConstants.ErrorMessages.JWT_PARSE_ERROR.getCode(),
+                    e.getMessage());
+        }
+        // Class cast exception is handled while parsing the idToken.
+        JWSAlgorithm expectedAlgorithm = (JWSAlgorithm) jwt.getHeader().getAlgorithm();
+        String jwkSourceUrl = null;
```

Comment:
Shall we use the StringUtils for defininf empty strings and also when checking them?

Reply:
Fixed 6646410ccf9035a93f8e161a8c67baed03d9b73e